torch/csrc/autograd/engine.cpp

```diff
@@ -605,10 +605,7 @@ void validate_outputs(
       grad = at::sum_to(std::move(grad), metadata.shape());
     }
 
-    bool input_is_complex = isComplexType(c10::typeMetaToScalarType(metadata.options().dtype()));
-    bool grad_is_complex = isComplexType(grad.scalar_type());
-
-    TORCH_CHECK(isFloatingType(grad.scalar_type()) || (input_is_complex == grad_is_complex));
```

Comment:
removed the check `(input_is_complex == grad_is_complex)` because this is not satisfied for `R -> C` functions

Reply:
It should be no? A real tensor should only have a real gradient, not a complex one right?

Reply:
yeah I confused it with `grad_out` so added it back again